util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+        if type(node_dict) == type(OrderedDict()):
+            return OrderedDict(new_tuples)
+        else:
+            return Dict(new_tuples)
+    else:
+        return node_dict
+
+
+def mcf_dict_rename_prop_value(node_dict: Union[dict, OrderedDict], prop: str,
+                               old_val: str,
+                               new_val: str) -> Union[dict, OrderedDict]:
+    """Helper function to rename a value of a given property of the Node.
+
+    Args:
+        node_dict: OrderedDict or Dict object representing a node in MCF file.
+        prop: Name of the property whose value needs to to be replaced.
+        old_val: Name of the value to be replaced.
+        new_val: New name of the value.
+        
+    Returns:
+        OrderedDict or Dict object representing a node in the MCF file.
+    """
+    if prop in node_dict:
+        for k, v in node_dict.items():
+            if k == prop and v['value'] == old_val:
+                v['value'] = new_val
+    return node_dict
+
+
+# wrapper function to rename namespace
+def mcf_dict_rename_namespace(node_dict: Union[dict, OrderedDict], old_ns: str,
+                              new_ns: str) -> Union[dict, OrderedDict]:
+    """Helper function to rename namespace of the Node.
+
+    Args:
+        node_dict: OrderedDict or Dict object representing a node in MCF file.
+        old_ns: Name of the namespace to be replaced.
+        new_ns: New name of the namespace.
+        
+    Returns:
+        OrderedDict or Dict object representing a node in the MCF file.
+    """
+    for k, v in node_dict.items():
+        if not (k == 'Node' and v['namespace'] == 'dcid'):
+            if v['namespace'] == old_ns:
+                v['namespace'] = new_ns
+    return node_dict
+
+
+def get_dcid_node(node_dict: Union[dict, OrderedDict]) -> str:
+    """Finds the dcid of the given Node.
+
+    Args:
+        node_dict: OrderedDict or Dict object representing a node in MCF file.
+        
+    Returns:
+        The dcid of the given Node, empty String if no dcid is found
+    """
+    if 'Node' in node_dict and node_dict['Node']['namespace'] == 'dcid':
+        return node_dict['Node']['value']
+    elif 'dcid' in node_dict:
+        return node_dict['dcid']['value']
+    else:
+        return ''
+
+
+# get dcid list from list of dict
+def get_dcids_node_list(node_list: list) -> list:
+    """Fetches the list of dcids for given list of nodes.
+
+    Args:
+        node_list: List of OrderedDict or Dict objects representing a node in MCF file.
+        
+    Returns:
+        The List dcids of the nodes present in the input list.
+    """
+    ret_list = []
+    for node_dict in node_list:
+        s = get_dcid_node(node_dict)
+        if s:
+            ret_list.append(s)
+    return ret_list
+
+
+# get dcid list subset of key values
+def get_dcids_prop_list(node_list: list, prop_list: list) -> list:
+    """Fetches the list of dcids for given list of nodes.
```

Comment:
pls add `that contain atleast one of given properties.`

Reply:
It checks for all the properties, added comment.